src/runtime/hexagon/hexagon_device_api.h

```diff
@@ -45,11 +45,27 @@ class HexagonDeviceAPI final : public DeviceAPI {
   static HexagonDeviceAPI* Global();
 
   //! \brief Constructor
-  HexagonDeviceAPI() {}
+  HexagonDeviceAPI() { hexbuffs = std::make_unique<HexagonBufferManager>(); }
 
   //! \brief Destructor
   ~HexagonDeviceAPI() {}
 
+  //! \brief Creates resource managers for the runtime
+  void AcquireResources() {
+    if (!hexbuffs->empty()) {
+      LOG(INFO) << "hexbuffs was not empty in AcquireResources";
```

Comment:
During my testing, I saw that there were allocations and frees before AcquireResources was called.  (During RPC initialization of this server). Sometimes due to timing, there were still allocations at this point.

It is not the latter.  I agree that we need a scheme for each user/client to have its own resource manager.

Reply:
I see --- so it's just that the resource manager is spinning up before the RPC initialization is totally done. So those are transient allocations that will (definitely?) be freed soon?

Reply:
You raise a good point - I'm going to investigate this more before proceeding.

I saw different behavior in the simulator vs real devices.  (Timing is different?)

Reply:
I verified that all items that were allocated before the call to "acquire" are indeed released by the time we release.

We do create the resources on the device side now, statically.  This is attempting to have the resource managers only created for the session, and then explicitly destroyed before the process is terminated.

Reply:
Would it make more sense to put `hexbuffs = std::make_unique<HexagonBufferManager>();` in this function (`AcquireResources`) rather than in the constructor for the device API.  This way, we know that `hexbuffs` is empty at the time we run "acquire" - no need to check.  It would also alleviate the need to check whether `hexbuffs` is empty on "release" and we could simply `hexbuffs.reset()`.

```
void AcquireResources() {
  CHECK_EQ(hexbuffs, nullptr);
  hexbuffs = std::make_unique<HexagonBufferManager>();
}

void ReleaseResources() {
  hexbuffs.reset();
  hexbuffs = nullptr;
}
```

The device api ctor could just set hexbuffs=nullptr.

Reply:
I have some pending updates that will keep a static object for allocations before and (potentially) after the runtime, and create/destroy the runtime manager in acquire/release.  Although not ideal, I think that's better.  The lifetime of runtime management is more clear.